common/src/main/java/com/microsoft/identity/common/internal/fido/CredManFidoManager.kt

```diff
@@ -23,28 +23,49 @@
 package com.microsoft.identity.common.internal.fido
 
 import android.content.Context
+import android.os.Build
 import androidx.credentials.CredentialManager
 import androidx.credentials.GetCredentialRequest
 import androidx.credentials.GetPublicKeyCredentialOption
 import androidx.credentials.PublicKeyCredential
+import androidx.credentials.exceptions.NoCredentialException
+import com.microsoft.identity.common.internal.providers.oauth2.WebViewAuthorizationFragment
+import com.microsoft.identity.common.java.flighting.CommonFlight
+import com.microsoft.identity.common.java.flighting.CommonFlightsManager
+import com.microsoft.identity.common.java.opentelemetry.AttributeName
+import com.microsoft.identity.common.logging.Logger
+import io.opentelemetry.api.trace.Span
 
 /**
  * Makes calls to the Android Credential Manager API in order to return an attestation.
  */
-class CredManFidoManager (val context: Context) : IFidoManager {
+class CredManFidoManager (val context: Context,
```

Comment:
can we use the OS version check to determine if CredManFidoManager or LegacyFido2ApiManager should be returned?

(no need to invoke CredManFidoManager and catch exception if we know for sure it's going to fail)

Reply:
(e.g. put it in a factory 


```kotlin
object FidoManagerFactory { // or put it in a companion object somewhere.
    fun getFidoManager() : IFidoManager {
      if (flightEnabled && os<=13){
            return LegacyFido2ApiManager()
      }

     return CredManFidoManager()
   }
}

```

Reply:
(Gave answer in one of your other comments)